### PR TITLE
#169: add pre and post hooks to canvas renderers

### PR DESCRIPTION
**Pull request by mdamien**
Following #180, but targeting `linkurious-version` instead of `master`

### CODE REVIEW

File: src/renderers/sigma.renderers.canvas.js [hunk 1]
```diff
@@ -102,8 +102,99 @@
     this.resize(false);
   };
 
+  /**
+   * Static method to render edges or nodes with the given renderers
+   *
+   * @param  {object}       params     The parameters passed in an object
+   *      {
+   *        renderers:           Renderers to be used
```
Comment (sheymann) [inline src/renderers/sigma.renderers.canvas.js]: is it an array of renderers?

File: src/renderers/sigma.renderers.canvas.js [hunk 2]
```diff
@@ -102,8 +102,99 @@
...
+   *        type:                "edges" or "nodes"
+   *        ctx:                 Canvas Context to draw on
+   *        settings:            Settings object to use
+   *        elements:            Elements to render
```
Comment (sheymann) [inline src/renderers/sigma.renderers.canvas.js]: array?

File: src/renderers/sigma.renderers.canvas.js [hunk 3]
```diff
@@ -102,8 +102,99 @@
...
 
+  /**
+   * Static method to render edges or nodes with the given renderers
+   *
+   * @param  {object}       params     The parameters passed in an object
+   *      {
+   *        renderers:           Renderers to be used
+   *        type:                "edges" or "nodes"
+   *        ctx:                 Canvas Context to draw on
+   *        settings:            Settings object to use
```
Comment (sheymann) [inline src/renderers/sigma.renderers.canvas.js]: function?

File: src/renderers/sigma.renderers.canvas.js [hunk 4]
```diff
@@ -102,8 +102,99 @@
...
+   *        elements:            Elements to render
+   *        graph:               Graph object (necessary for edge rendering)
+   *        start:    (optional) Starting index of the elements to render
+   *        end:      (optional) Last index of the elements to render
+   *      }
+   */
+  sigma.renderers.canvas.applyRenderers = function(params) {
+    var i,
+        renderer,
+        specialized_renderer,
```
Comment (sheymann) [inline src/renderers/sigma.renderers.canvas.js]: style: use camelCase like the rest of the source

File: src/renderers/sigma.renderers.canvas.js [hunk 5]
```diff
@@ -102,8 +102,99 @@
...
+    }
+  };
 
 
+  /**
+   * Render a batch of edges
+   *
+   * @param    {integer}               start   Starting index of the edges
+   *                                           to render
+   * @param    {integer}               end     An object of options.
```
Comment (sheymann) [inline src/renderers/sigma.renderers.canvas.js]: wrong comment

File: src/renderers/sigma.renderers.canvas.js [hunk 6]
```diff
@@ -102,8 +102,99 @@
...
+   * @param    {object}                options An object of options.
```
Comment (sheymann) [inline src/renderers/sigma.renderers.canvas.js]: options or settings?

**Comment by mdamien** [comment]
Close PR to do it with the new workflow #201

**Pull request closed without merging**